File: src/main/java/model/organization/event/AssignmentEventFactory.java
package model.organization.event;

import model.organization.relation.Assignment;

/**
 * The {@linkplain AssignmentEventFactory} interface defines the API for constructing {@linkplain AssignmentEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface AssignmentEventFactory {

	/**
	 * Constructs an {@linkplain AssignmentEvent}.
	 *
	 * @param category
	 *            the {@linkplain EventCategory}.
	 * @param assignment
	 *            the {@linkplain Assignment}.
	 *
	 * @return a {@linkplain AssignmentEvent}.
	 */
	AssignmentEvent build(EventCategory category, Assignment assignment);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/AssignmentEventFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.relation.Assignment
import runtimemodels.chazm.model.event.EventType

/**
 * The [AssignmentEventFactory] interface defines the API for constructing [AssignmentEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface AssignmentEventFactory {
    /**
     * Constructs an [AssignmentEvent].
     *
     * @param category   the [EventType].
     * @param assignment the [Assignment].
     * @return a [AssignmentEvent].
     */
    fun build(category: EventType, assignment: Assignment): AssignmentEvent
}
